### ISSUE TITLE
GLONASS ephemeris conversion PZ-90 -> WGS84

### ISSUE THREAD

**Issue by borax00** [tomojitakasu/RTKLIB#676]
Hi,

I've been going through the libRTK source code.  Where the GLONASS satellite-position is caluclated from the broadcast ephemeris, it seems to be missing the conversion from PZ-90 to WGS84 (per e.g. Section 3.1.3.2 of https://gssc.esa.int/navipedia/GNSS_Book/ESA_GNSS-Book_TM-23_Vol_I.pdf).
Is this an omission or am I missing something?

**Comment by yydgis** [comment]
currently, PZ-90 closely matches WGS84/ITRF, just ignore the difference

**Comment by tomojitakasu** [comment]
The errors due to the coordinate systems' differfence, i.e. PZ90 (even as PZ90.02) and WGS84, are usually negligible compared to the errors of GLONASS broadcast ephemeris.

Refer the manual E.2.

https://www.rtklib.com/prog/manual_2.4.2.pdf

**Comment by borax00** [comment]
Awesome, thanks for the reply.

**Comment by ValZapod** [comment]
> PZ-90 closely matches WGS84/ITRF, just ignore the difference

No anymore, USA GNSS (GPS) migrated to G2139 on 3 January 2021, which is aligned with ITRF2014. PZ-90.11 (used since 2013 in GLONASS) is still on quite old ITRF2008.

But in April 2020 ITRF2020 was released, so...

**Comment by Waley04** [comment]
> Hi,
> 
> I've been going through the libRTK source code. Where the GLONASS satellite-position is caluclated from the broadcast ephemeris, it seems to be missing the conversion from PZ-90 to WGS84 (per e.g. Section 3.1.3.2 of https://gssc.esa.int/navipedia/GNSS_Book/ESA_GNSS-Book_TM-23_Vol_I.pdf). Is this an omission or am I missing something?

This error has a negligible effect on the solution.

**Comment by ValZapod** [comment]
> negligible effect on the solution.

Again, not ever since GPS migrated to G2139.

**Comment by Waley04** [comment]
I agree. Again, for the sake of positioning and navigation, the G2139 is coincident with ITRF14(or IGb14).

**Comment by ValZapod** [comment]
> G2139 is coincident with ITRF14

Different default epoch, different epoch it started to actually work on GPS and a different ellipsoid. GLONASS is much worse of an deviation though. The fact of the matter proj can very simply convert GLONASS to ITRF2008 (see presentation on wikipedia page of GLONASS and needed EPSG code for a tansformation on epsg.org) and that can be converted to ITF2014.

**Comment by ValZapod** [comment]
Meanwhile GPS moved to IGS20, aligned with ITRF2020.
